pkg/controller/bundlec/resource_sync_task.go

```diff
@@ -717,6 +718,15 @@ func (st *resourceSyncTask) updateResource(resClient dynamic.ResourceInterface,
 	if err != nil {
 		return nil, false, errors.Wrap(err, "specification check failed")
 	}
+
+	// Delete the DeletionTimestamp annotation if it is present
+	annotations := updated.GetAnnotations()
```

Comment:
@ash2k is this the correct place to put the annotation deletion code?

Reply:
Made a note here https://github.com/atlassian/smith/issues/289#issuecomment-452577218

Reply:
@ash2k actually I would need to remove the annotation from `actual`, rather than from `spec` (AFAIK the "merge" logic will combine annotations from `actual` and `spec` => will keep the annotation in the result), and removing annotation from `actual` seems hacky to me (we shouldn't mutate `actual` before comparison?).
Mutating `updated` is fine (the merge result, "desired state"), on the other hand.
So I think I would prefer to keep the code unless I misunderstood something